appserver/security/jacc.provider.file/src/main/java/com/sun/enterprise/security/provider/PolicyConfigurationImpl.java

```diff
@@ -972,8 +973,10 @@ private boolean stateIs(int stateValue) {
         boolean inState = _stateIs(stateValue);
         
         if (stateValue == INSERVICE_STATE && !inState) {
-            if (fileArrived(true) || fileArrived(false)) {
-
+            Boolean fileArrived = AccessController.doPrivileged(
+                    (PrivilegedAction< Boolean>) () -> fileArrived(true) || fileArrived(false)
+            );
```

Comment:
I'd expect `initialize` below to also require same identity (effectively expanding privileged to entire body.

But since you've tested it works this way, the only nitpick would be space after `<`. :)

Reply:
The tests succeeded with fileArrived covered only. 
Formatting is fixed (together with year of the copyright).